--- /usr/lib/python3/dist-packages/mesonbuild/compilers/d.py.orig       2026-06-11 00:04:55.928476177 +0000
+++ /usr/lib/python3/dist-packages/mesonbuild/compilers/d.py    2026-06-11 00:05:39.589738384 +0000
@@ -167,6 +167,12 @@
             return []
         return ['-fPIC']
 
+    def get_pie_args(self):
+       return ['--relocation-model=pic']
+
+    def get_pie_link_args(self):
+        return ['-L-pie']
+
     def get_optimization_link_args(self, optimization_level: str) -> T.List[str]:
         if optimization_level != 'plain':
             return self._get_target_arch_args()
--- /usr/lib/python3/dist-packages/mesonbuild/dependencies/dub.py.orig  2026-06-11 00:06:36.121409978 +0000
+++ /usr/lib/python3/dist-packages/mesonbuild/dependencies/dub.py       2026-06-11 00:07:12.687730323 +0000
@@ -119,7 +119,8 @@
                    '{!r}'.format(name, self.dubbin.get_path()))
 
         # we need to know the target architecture
-        dub_arch = self.compiler.arch
+        host_machine = self.env.machines.host
+        dub_arch = f'{host_machine.cpu_family}-linux-{host_machine.system}'
 
         # we need to know the build type as well
         dub_buildtype = str(environment.coredata.optstore.get_value_for(OptionKey('buildtype')))
